graylog2-web-interface/src/views/components/contexts/DashboardPageContextProvider.tsx

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import { useState, useEffect, useMemo, useCallback } from 'react';
+import { useLocation, useHistory } from 'react-router-dom';
+import URI from 'urijs';
+
+import useQuery from 'routing/useQuery';
+import DashboardPageContext from 'views/components/contexts/DashboardPageContext';
+import { ViewActions } from 'views/stores/ViewStore';
+import { ViewStatesStore } from 'views/stores/ViewStatesStore';
+import { useStore } from 'stores/connect';
+
+const _clearURI = (query) => new URI(query)
+  .removeSearch('page');
+
+const _updateQueryParams = (
+  newPage: string | undefined,
+  query: string,
+) => {
+  let baseUri = _clearURI(query);
+
+  if (newPage && newPage !== 'new') {
```

Comment:
We can now remove the check `newPage !== 'new'`

Reply:
@kmerz I saw that you removed the line `if (newPage && newPage !== 'new') {` completely. I prefer to keep `if (newPage) {` because we are calling this function when calling `unSetDashboardPageParam`, which means `newPage` can be `undefined`. Currently we would add the URL query param `page` without any value in this case.